homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,251 @@
+                    SUPPORT_PREVIOUS_TRACK | SUPPORT_VOLUME_STEP | \
+                    SUPPORT_TURN_ON | SUPPORT_PAUSE | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_PREFER_PICON, default=DEFAULT_PREFER_PICON): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of an enigma2 media player."""
+    if discovery_info:
+        discovered_config = {
+            CONF_NAME: DEFAULT_NAME,
+            # Do not use the discovered port, as it is the streaming
+            # service port (8001) which is not what we want.
+            CONF_PORT: DEFAULT_PORT,
+            CONF_HOST: discovery_info['host'],
+            CONF_USERNAME: DEFAULT_USERNAME,
+            CONF_PASSWORD: DEFAULT_PASSWORD,
+            CONF_SSL: DEFAULT_SSL,
+            CONF_PREFER_PICON: DEFAULT_PREFER_PICON,
+        }
+        add_devices([Enigma2Device(discovery_info['hostname'],
+                                   discovered_config)], True)
+        return
+
+    add_devices([Enigma2Device(config[CONF_NAME], config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = None
+        self._name = name
+        self.e2_box = \
+            openwebif.api.CreateDevice(host=config[CONF_HOST],
+                                       port=config[CONF_PORT],
+                                       username=config[CONF_USERNAME],
+                                       password=config[CONF_PASSWORD],
+                                       is_https=config[CONF_SSL],
+                                       prefer_picon=config[CONF_PREFER_PICON])
+
+        self.volume = 10
+        self.current_service_channel_name = None
+        self.current_programme_name = None
+        self.current_service_ref = None
+        self.muted = False
+        self.picon_url = None
+        self.status_info = {}
+        self.sources = self.e2_box.sources
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_ENIGMA2
+
+    def turn_off(self):
+        """Turn off media player."""
+        self.e2_box.toggle_standby()
```

Comment:
Are there no discrete turn_off and turn_on commands available?

Reply:
No but I will add thos to the pip module.

Reply:
Okay, I'm not sure it makes any difference if that's not possible, but your choice!

Reply:
Done now in the pypi module.